neqo-http3/src/hframe.rs

```diff
@@ -105,20 +106,18 @@ impl HFrame {
     }
 }
 
-#[derive(Copy, Clone, PartialEq, Debug)]
+#[derive(Clone, Debug)]
 enum HFrameReaderState {
-    BeforeFrame,
-    GetType,
-    GetLength,
-    GetData,
-    UnknownFrameDischargeData,
-    Done,
+    BeforeFrame { decoder: IncrementalDecoder },
+    GetType { decoder: IncrementalDecoder },
+    GetLength { decoder: IncrementalDecoder },
+    GetData { decoder: IncrementalDecoder },
+    UnknownFrameDischargeData { decoder: IncrementalDecoder },
```

Comment:
It seems like this should be a struct that holds a decoder and an enum rather than an enum with the same type attached to every alternative.

Reply:
I want to split IncrementalDecoder into 3 decoders, I did not get to it yesterday. So each enum will hold a different decoder.

Reply:
I see what you mean.  Yeah, that's a good idea.